### PR TITLE
Add MLIR conv3d support

### PR DESCRIPTION
**Pull request by turneram**

### CODE REVIEW

File: src/targets/gpu/fuse_mlir.cpp [hunk 1]
```diff
@@ -257,7 +257,7 @@ auto is_mlir_conv(mlir_mode mode)
         value v    = ins->get_operator().to_value();
         auto group = v.at("group").to<int>();
         // Avoid MLIR assertion: Index < Length && "Invalid index!"
-        if(ins->get_shape().lens().size() != 4)
+        if(ins->get_shape().lens().size() != 4 and group > 1)
```
[the remark this answers is not in the document]
Comment (apwojcik) [reply src/targets/gpu/fuse_mlir.cpp:260]: This works great! Let's go with it and merge :-)

**Review by apwojcik — APPROVED** [review]

File: src/targets/gpu/fuse_mlir.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (umangyadav) [reply src/targets/gpu/fuse_mlir.cpp:260]: @apwojcik  do you have data points for that ?

Comment (umangyadav) [inline src/targets/gpu/fuse_mlir.cpp:260]: @turneram  @apwojcik  can you please make this change enable only on windows for now ?

**Review by umangyadav — APPROVED** [review]
Approving but we need to collect data on performance.

File: src/targets/gpu/fuse_mlir.cpp (hunk 1 above)
Comment (apwojcik) [reply src/targets/gpu/fuse_mlir.cpp:262]: done

**Pull request merged**